tests/src/test/scala/org/http4s/parser/ContentDispositionHeaderSuite.scala

```diff
@@ -41,4 +40,11 @@ class ContentDispositionHeaderSuite extends Http4sSuite {
     )
   }
 
+  test("Content-Disposition header should pick 'filename*' parameter and ignore 'filename'") {
+    val headerValue = """form-data; filename="filename"; filename*=UTF-8''http4s%20filename"""
```

Comment:
Actually, 'filename*' should contain non-ISO-8859-1 characters in terms of fairness, but [that fix](https://github.com/http4s/http4s/pull/6473) must be merged first.